File: module_ssh/src/main/java/com/xiaoyv/ssh/main/model/SshListModel.java
package com.xiaoyv.ssh.main.model;

import com.xiaoyv.busines.room.database.DateBaseManger;
import com.xiaoyv.busines.room.entity.SshEntity;
import com.xiaoyv.ssh.main.contract.SshListContract;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;

/**
 * SshListModel
 *
 * @author why
 * @since 2020/11/29
 **/
public class SshListModel implements SshListContract.Model {

    @Override
    public Observable<List<SshEntity>> p2mQueryLocalSsh() {
        return Observable.create(emitter -> {
            emitter.onNext(DateBaseManger.get().getSshDao().getAll());
            emitter.onComplete();
        });
    }

    @Override
    public Observable<List<SshEntity>> p2mQueryLocalSshByGroup(String group) {
        return Observable.create(emitter -> {
            emitter.onNext(DateBaseManger.get().getSshDao().getAllByGroup(group));
            emitter.onComplete();
        });
    }
}

File: module_ssh/src/main/java/com/xiaoyv/ssh/main/SshListModel.kt
package com.xiaoyv.ssh.main

import com.xiaoyv.blueprint.exception.RxException
import com.xiaoyv.business.room.database.DateBaseManger
import com.xiaoyv.business.room.entity.SshEntity

import io.reactivex.rxjava3.core.Observable

/**
 * SshListModel
 *
 * @author why
 * @since 2020/11/29
 */
class SshListModel : SshListContract.Model {
    override fun p2mQueryLocalSsh(): Observable<List<SshEntity>> {
        return Observable.create {
            it.onNext(DateBaseManger.get().sshDao.queryAll())
            it.onComplete()
        }
    }

    override fun p2mQueryGroup(): Observable<List<String>> {
        return Observable.create {
            // 数据库内容
            val groupList = DateBaseManger.get().sshDao.queryGroup().toMutableList()
            if (groupList.isNullOrEmpty()) {
                it.onError(RxException("rdp record list is null!"))
                return@create
            }

            it.onNext(groupList)
            it.onComplete()
        }
    }

    override fun p2mDeleteSsh(sshEntity: SshEntity): Observable<Boolean> {
        return Observable.create {
            DateBaseManger.get().sshDao.delete(sshEntity)
            it.onNext(true)
            it.onComplete()
        }
    }

    override fun p2mQueryLocalSshByGroup(group: String): Observable<List<SshEntity>> {
        return Observable.create {
            it.onNext(DateBaseManger.get().sshDao.queryByGroup(group))
            it.onComplete()
        }
    }
}